doc/whats_new/v1.1.rst

```diff
@@ -645,6 +645,9 @@ Changelog
   `fit` instead of `__init__`.
   :pr:`21434` by :user:`Krum Arnaudov <krumeto>`.
 
+- |Fix| :meth:`preprocessing.FunctionTransformer._check_inverse_transform`
+  now provides informative error message when input has mixed dtype. :pr:`19916` by
+  :user:`Zhehao Liu <MaxwellLZH>`.
```

Comment:
Since the user guide is public facing, it's better to reference public interfaces:

```suggestion
- |Fix| :class:`preprocessing.FunctionTransformer` with `check_inverse=True`
  now provides informative error message when input has mixed dtypes. :pr:`19916` by
  :user:`Zhehao Liu <MaxwellLZH>`.
  
```

(Also a newline after the changelog entry)